File: 백준/Silver/11637. 인기 투표/인기 투표.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            int count = Integer.parseInt(br.readLine());

            Map<Integer, Integer> map = new HashMap<>();
            int maxVote = 0;

            // 가장 큰 투표 수 구하기
            int totalVotes = 0;
            for (int i = 1; i <= count; i++) {
                int vote = Integer.parseInt(br.readLine());
                if (vote > maxVote) {
                    maxVote = vote;
                }
                totalVotes += vote;
                map.put(i, vote);
            }

            // 가장 큰 투표 수와 같은 사람 구하기
            List<Integer> answer = new ArrayList<>();
            int finalMaxVote = maxVote;
            map.forEach((key, value) -> {
                if (value == finalMaxVote) {
                    answer.add(key);
                }
            });

            int half = totalVotes/2 + 1;
            if (answer.size() == 1 && maxVote >= half) {
                System.out.println("majority winner " + answer.get(0));
            } else if (answer.size() == 1) {
                System.out.println("minority winner " + answer.get(0));
            } else {
                System.out.println("no winner");
            }
        }
    }
}

File: 백준/Silver/11637. 인기 투표/인기 투표.py
t = int(input())
for _ in range(t):
    n = int(input())
    arr = [int(input()) for i in range(n)]

    sum_votes = sum(arr)
    max_vote = max(arr)

    half = sum_votes//2+1
    if arr.count(max_vote) == 1 and half <= max_vote:
        print("majority winner", arr.index(max_vote)+1)
    elif arr.count(max_vote) == 1:
        print("minority winner", arr.index(max_vote)+1)
    else:
        print("no winner")
